issue 514910

Zer00Player opened the pull request.

Zer00Player committed:
issue 514910

SugarHedgehog commented on zdn/matege2024b/21/514910.js:
```diff
@@ -0,0 +1,34 @@
+    retryWhileError(function () {
+        NAinfo.requireApiVersion(0, 2);
+        let numberOfCut, numberPiecesOfWood;
+        do {
```
Эта конструкция не имеет особого смысла, хотя и рабочая
Можно, например, сначала определить `numberOfCut`, а следующим действием определить `numberPiecesOfWood  = sl(numberOfCut+1, 200)`

При этом стоит сделать `numberOfCut` чуточку меньше максимального значения в `sl()`

P.S. `sl()=sluchch()` ДА, можно писать просто `sl`!

SugarHedgehog commented on zdn/matege2024b/21/514910.js:
```diff
@@ -0,0 +1,34 @@
+        let questionNumerAdverbs = ['Сколько', 'Какое количество'].iz();
+
+        let startText;
+        if (begginingAdverbs[0] === ' ') {
```
Схема рабочая
Но я бы обратила внимание на https://learn.javascript.ru/ifelse#uslovnyy-operator

SugarHedgehog commented on zdn/matege2024b/21/514910.js:
```diff
@@ -0,0 +1,34 @@
+
+        let startText;
+        if (begginingAdverbs[0] === ' ') {
+            startText = verbTakeVariants[0].toZagl();
```
Замечание, у вас дважды повторяется `.toZagl()` в действиях - это нехорошо
 `.toZagl()` можно спокойно использовать в стр. 27

SugarHedgehog commented on zdn/matege2024b/21/514910.js:
```diff
@@ -0,0 +1,34 @@
+            numberPiecesOfWood = sluchch(3, 200, 1);
+        } while (numberOfCut >= numberPiecesOfWood);
+
+        let begginingAdverbs = ['изначально', 'в начале', ' сначало', ' '].iz(2);
```
Перед `сначало` пробел - из-за этого не сработает правильно `.toZagl()` далее, тк сделает пробел(первый символ строки) заглавным)

SugarHedgehog reviewed (commented):
![изображение](https://github.com/user-attachments/assets/bda996d0-1528-4d79-9c51-72ad91e5b405)
![изображение](https://github.com/user-attachments/assets/cf673d56-19f4-4b2d-840a-00d84eff11d6)
![изображение](https://github.com/user-attachments/assets/e9667211-7966-4a69-82dd-300257788d46)

1) попадаются вот такие несвязные формулировки
2) ответ сходиться
3) немного подлатать

nickkolok commented:
Алсо, сначалА.

И да, не стОит пугаться плашки о "конфликтах", в данном случае это ерунда.

Zer00Player committed:
issue 514910 edit 1

SugarHedgehog reviewed (commented):
![изображение](https://github.com/user-attachments/assets/3ea001fc-cc4f-4852-a836-aaf8d1715193)
Если рандомом `numberOfCut` приравняется к 200
То будет такой баг)
_На миллион первый раз вылезет - знаем мы этот рандом_

SugarHedgehog commented:
>  плашки о "конфликтах"

Да, это забота принимающего

Zer00Player committed:
issue 514910 edit 2

SugarHedgehog committed:
Merge branch 'devel' into newtask-1897

SugarHedgehog committed:
Fix indent

Понятия не имею, почему оно при решении конфликта показывает, что всё ровненько, а при сохранении съезжает
